File: src/types/Link.ts
export interface NoteLink {
  id: string;
  source_id: string;
  target_id: string;
  link_type: LinkType;
  label?: string;
  color?: LinkColor;
  directional?: boolean;
  created_at: string;
}

export type LinkType = 
  | 'Related'
  | 'Reference'
  | 'FollowUp'
  | 'Contradicts'
  | 'Supports'
  | { Custom: string };

export type LinkColor = 'purple' | 'yellow';

export interface LinksDatabase {
  links: NoteLink[];
}

File: src-tauri/src/models/link.rs
use serde::{Deserialize, Serialize};
use chrono::{DateTime, Utc};

#[derive(Serialize, Deserialize, Clone)]
pub struct NoteLink {
    pub id: String,
    pub source_id: String,
    pub target_id: String,
    pub link_type: LinkType,
    pub label: Option<String>,
    pub color: Option<LinkColor>,
    pub directional: Option<bool>,
    pub created_at: DateTime<Utc>,
}

#[derive(Serialize, Deserialize, Clone)]
pub enum LinkType {
    Related,
    Reference,
    FollowUp,
    Contradicts,
    Supports,
    Custom(String),
}

#[derive(Serialize, Deserialize, Clone)]
pub enum LinkColor {
    Purple,
    Yellow,
}